src/User/UserContainer.jsx

```diff
@@ -64,7 +64,7 @@ const UserContainer = () => {
   return (
     <Switch>
       <Route path="/users/:id/edit">
-        <UpdateProfile />
+        {isCurrentUser ? <UpdateProfile /> : <Redirect to="/" />}
```

Comment:
I would probably suggest redirecting to their profile instead of sending them to the home page